### ISSUE TITLE
Correct installation of data files

### ISSUE THREAD

**Issue by jgosmann** [nengo/nengo#708]
With the fix of #704  in #706 I discovered problems relating the installation of data files.

## Problem 1

Examples are not installed, but `tox` runs unit tests on the installed package. Thus the unit test for running the examples fails. What is the best way to handle this?

* Install examples to always be able to run the complete test suite? If so, there seem to be two possibilities:
  1. Install them as a Python package. That would require to add `__init__.py` files. Also we should probably rename the directory to something like `nengo-examples` or move it into the `nengo` directory.
  2. Install them as data files. That way they go into some directory specified by us below `sys.prefix` (as far as I found out, it's hard to navigate the documentation for all the different Python setup packages).
* Don't install examples. How should this handled by test run in tox?
  * Just fail? But then one has to look closely at the tox output to know whether there was a problem.
  * Just succeed if the examples can't be found? But then `tox` might report succeeding tests even if some of the examples fail.
  * Add an alternate way to locate the tests? It could be possible to add an environment variable or py.test command line option which is set in `tox.ini` to the directory with the (not installed) examples.

`nengo.utils.paths` defines an `example_dir` which is wrong for any installed Nengo. It only works for uninstalled (and maybe develop installs).

## Problem 2

`nengorc` is currently installed as a data file in `sys.prefix/nengo/nengorc`, which is not the path determined by `nengo.utils.paths`. Should we continue to install it as a data file? In either case the path has to be fixed.

**Comment by hunse** [comment]
I don't think we want to install them as part of the python package. Then they'd be in some out-of-the-way folder like /usr/lib/python2.7/dist-packages/nengo/examples or something. I'd install them as data files, so that users can go and run them themselves if they want (i.e., they don't have to install from source to get the examples).

**Comment by tbekolay** [comment]
Summary of discussion from dev meeting:

This is a problem! We want people to have easy access to the example files, but it's not clear how that can be achieved with `setup.py`. Mainly because `setup.py` can install in system-wide or user-level locations, and while it's obvious where those are for Python packages, data files aren't quite as obvious. You can imagine a situation where a different user has installed Nengo, but another user is using it and wants access to the examples, including the ability to run them (so, it needs to be somewhere that that user has write-access to). I think Jan enumerated the possible options well above... some additional options that were discussed: 1) don't install examples, but make a .zip of all the examples available somewhere obvious; 2) install examples as data files, and have a function like `nengo.copy_examples('~/nengo_scripts')` to copy them somewhere; 3) pull a MATLAB and have Nengo create a folder in a user's home directory and fill it with examples either automatically (on import) or on request.

No consensus on a solution. If better solutions jump out to you please suggest it here!

Another related problem, but probably for future-us to worry about: we'll want to ship these examples in a form that's compatible with the Nengo GUI when that's being packaged up. Possible ways to do this include:

1. Maintaining a separate set of `.py` scripts that are translated from the notebooks manually, for ultimate flexibility.
2. Generating a set of `.py` scripts from the examples using a custom script to strip out things that would be problematic for the GUI.
3. Generating a set of `.py` scripts from the examples, and having the GUI automatically deal with the bits that would be problematic (essentially, constructing a simulator object).

**Comment by jgosmann** [comment]
In nengo_spa I setteled on installing the the examples as package data (for now). By adding an appropriate entry point to `setup.py` it should allow the GUI to retrieve those additional examples. nengo_spa also provides a command (nengo/nengo_spa#49) to copy these examples to a user defined location.

This is somewhat similar to [how an IPython kernel is installed](http://ipython.readthedocs.io/en/stable/install/kernel_install.html) which also requires some files to be installed to a user directory.
